### PR TITLE
Update h264_pusher.c

### PR DESCRIPTION
**Pull request by Dw9**
修复win构建失败 strcasestr->strstr

### CODE REVIEW

File: api/tests/h264_pusher.c [hunk 1]
```diff
@@ -68,7 +68,7 @@ void on_push_shutdown(void *user_data, int err_code, const char *err_msg) {
 void API_CALL on_regist(void *user_data, mk_media_source sender, int regist) {
     Context *ptr = (Context *)user_data;
     const char *schema = mk_media_source_get_schema(sender);
-    if (strcasestr(ptr->url, schema) != ptr->url) {
+    if (strstr(ptr->url, schema) != ptr->url) {
```
Comment (suzp1984) [inline api/tests/h264_pusher.c:71]: strstr is case-sensitive.

Comment (suzp1984) [inline api/tests/h264_pusher.c:71]: strstr is case-sensitive.

```
const char *strcasestr(const char *big, const char *little){
    string big_str = big;
    string little_str = little;
    strToLower(big_str);
    strToLower(little_str);
    auto pos = strstr(big_str.data(), little_str.data());
    if (!pos){
        return nullptr;
    }
    return big + (pos - big_str.data());
}
```
https://github.com/ZLMediaKit/ZLToolKit/blob/04d1c47d2568f5ce1ff84260cefaf2754e514a5e/src/Util/util.cpp#L280-L290
There is a polyfill for _WIN32, `ZLToolKit` is inside the `3rdpart` folder, can be included in this `c` code.
According to the _WIN32 doc, there is an impl `StrStrIA`, I don't know why `ZLToolKit` gives its own impl.
https://learn.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-strstria

Anyway, I think use `ZLToolKit`'s utils can be better.

Comment (Dw9) [reply api/tests/h264_pusher.c:71]: schema is originally lowercase, "strcasestr" may be designed for 'poka yoke',avoid user input RTM/SP
This's just a  test case,  we can keep it simple

Comment (suzp1984) [reply api/tests/h264_pusher.c:71]: OK, If Simplicity is the first principle here.

**Pull request merged**